### PR TITLE
Implement DPI scaling for Windows

### LINKED ISSUE #6735: Default DPI on Windows is wrong, affects HUD scaling
Here's the bug in question, http://i.imgur.com/o8I7ytZ.png and https://jordach.net/images/ppxdgexmfhcvlnajtl.png (provided by @benrob0329 )

Here's what it should look like, https://jordach.net/images/mdosditzrdlduukpl.png

About 10 months ago, @sfan5, has found the source of the issue, and is related to porting.h.

```
<sfan5> >float size_factor = m_hud_scaling * porting::getDisplayDensity();
<sfan5> i'm guessing this is at fault
<sfan5> its probably unimpl. (=1) on win
```

The line in question, is found here:

https://github.com/minetest/minetest/blob/master/src/hud.cpp#L388

The issue also extends to pixel offsets, which should be 1:1 as the lua_api.txt implies, not 1:4, depending on display density, as these lines;

https://github.com/minetest/minetest/blob/master/src/hud.cpp#L389-L392

After replacing RenderingEngine::getDisplayDensity() with a 1 (according to IRC logs), it then looks corrected, as shown here:

https://kitsunemimi.pw/i/c2a30f175559ac81.png

This affects all versions of Minetest from 0.4.7 through 0.5-dev (ab947bd)

However, a commit by @sapier, implies that it may originate there: https://github.com/minetest/minetest/commit/1838a3fd696782b1733a435bbb25accf3e40d1f3

Related bugs; https://github.com/minetest/minetest/issues/6296

### PR DESCRIPTION
**Pull request by sfan5**
Reads DPI from OS instead of assuming 72dpi as default.
This makes GUI, HUD and font size consistent with Linux.

Tested in Wine and on Windows 8.1.
Note that the code doesn't use [`GetDpiForWindow`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdpiforwindow) because that is only implemented in Windows 10.
fixes #6735

## To do

This PR is a Ready for Review.

### CODE REVIEW

**Comment by sfan5** [comment]
fixes #6735

**Comment by Wuzzy2** [comment]
@sorcerykid: This one might be interesting for you.

File: src/client/renderingengine.cpp [hunk 1]
```diff
@@ -683,12 +696,42 @@ float RenderingEngine::getDisplayDensity()
...
+		float dpi = GetDeviceCaps(hdc, LOGPIXELSX);
+		ReleaseDC(hWnd, hdc);
+		return dpi / 96.0f;
+	}
+
+	/* return manually specified dpi */
+	return g_settings->getFloat("screen_dpi") / 96.0f;
+}
+
+float RenderingEngine::getDisplayDensity()
```
Comment (nerzhul) [inline src/client/renderingengine.cpp:716]: i don't think we should cache this, what happen when you change minetest window to another screen ?

Comment (nerzhul) [reply src/client/renderingengine.cpp:716]: or we may have a function to detect the screen is not the old one and retrigger the calculation

Comment (sfan5) [reply src/client/renderingengine.cpp:716]: Runtime DPI changes require a [Window 10-only API](https://docs.microsoft.com/de-de/windows/win32/api/windef/ne-windef-dpi_awareness) and probably also needs to be implemented in Irrlicht. (and our GUI code might not be laid out for it)

Other platforms do not support that yet either.

Comment (nerzhul) [reply src/client/renderingengine.cpp:716]: i don't know for GL rendering but on my work laptop i have 4k DPI and external screens and gnome apps are managing it properly. If you don't think it's possible, just ignore my comment

Comment (sfan5) [reply src/client/renderingengine.cpp:716]: It's definitely possible, but it should be implemented some other time in a different PR.

Comment (nerzhul) [reply src/client/renderingengine.cpp:716]: okay fine to me

**Pull request merged**